### ISSUE TITLE
Null Coalescing Assignment: Compiler should not report IDE0052 when assigning a value to a property backing field

### ISSUE THREAD

**Issue by WilStead** [dotnet/roslyn#32842]
**Version Used**: Visual Studio 2019 Preview 2

**Steps to Reproduce**:
When using the new null coalescing assignment operator to initialize a field used as the backing store for a property, message IDE0052 is reported for the backing field.

For example:
```c#
private Foo _foo1; // Message IDE0052: Private member '_foo1' can be removed as the value assigned to it is never read.
public Foo Foo1 => _foo1 ??= new Foo();

private Foo _foo2; // No message when using this kind of conditional assignment construction
public Foo Foo2 => _foo2 ?? (_foo2 = new Foo());
```

**Expected Behavior**:
No message is reported.

**Actual Behavior**:
Message IDE0052 is reported.
